c7n/resolver.py

```diff
@@ -40,12 +41,27 @@ def resolve(self, uri):
         else:
             # TODO: in the case of file: content and untrusted
             # third parties, uri would need sanitization
-            fh = urlopen(uri)
-            contents = fh.read().decode('utf-8')
-            fh.close()
+            req = Request(uri, headers={"Accept-Encoding": "gzip"})
+            response = urlopen(req)
+            contents = self.handle_content_encoding(response)
+            response.close()
+
         self.cache.save(("uri-resolver", uri), contents)
         return contents
 
+    def handle_content_encoding(self, response):
+        if response.info().get('Content-Encoding') != 'gzip':
+            return response.read().decode('utf-8')
+
+        content = gzip.decompress(response.read())
+        decomp_req = content.splitlines()
+        contents = ''
+        for line in decomp_req:
+            contents += line.decode('utf-8')
```

Comment:
why are you decoding and reassembling the string line by line instead of once?

Reply:
Updated.